federation/pkg/federatedtypes/adapter.go

```diff
@@ -54,6 +54,11 @@ type FederatedTypeAdapter interface {
 	ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error)
 
 	NewTestObject(namespace string) pkgruntime.Object
+
+	// Individual adapter impl are supposed to provide the hook functions.
+	// Supposed to be nil on those adapters which don't need them (one or both).
+	GetScheduleHook() ScheduleHook
```

Comment:
I realize I should have explained my rational for the ``IsScheduler`` method.  If we accidentally screw up the signature of an interface method (e.g. ``Schedule``), a cast would fail in the same way as for a type that didn't implement the method at all.  So the ``isScheduler`` method effectively takes the place of the type check that go would perform if the type were passed as the interface.